helpers/helpers.go

```diff
@@ -153,6 +156,75 @@ func GetIncludedBundles(filename string) ([]string, error) {
 	return includes, nil
 }
 
+// UnpackFile unpacks a .tar or .tar.gz/.tgz file to a given directory.
+// Should be roughly equivalent to "tar -x[z]f file -C dest". Does not
+// overwrite; returns error if file being unpacked already exists.
+func UnpackFile(file string, dest string) error {
+	fr, err := os.Open(file)
+	if err != nil {
+		return err
+	}
+	defer func() {
+
+		_ = fr.Close()
+	}()
+
+	var tr *tar.Reader
+
+	// If it's a compressed tarball
+	if strings.HasSuffix(file, ".tar.gz") || strings.HasSuffix(file, ".tgz") {
+		gzr, err := gzip.NewReader(fr)
+		if err != nil {
+			return errors.Wrapf(err, "Error decompressing tarball: %s", file)
+		}
+		defer func() {
+			_ = gzr.Close()
+		}()
+		tr = tar.NewReader(gzr)
+	} else {
+		tr = tar.NewReader(fr)
+	}
+
+	for {
+		hdr, err := tr.Next()
+		if err == io.EOF {
+			// end of archive
+			break
+		} else if err != nil {
+			return errors.Wrapf(err, "Error reading contents of tarball: %s", file)
+		}
+
+		// Skip GitHub generated "extended header" file
+		if strings.Compare(hdr.Name, "pax_global_header") == 0 {
+			continue
+		}
+
+		out := filepath.Join(dest, hdr.Name)
+
+		switch hdr.Typeflag {
+		case tar.TypeDir:
+			if _, err := os.Stat(out); err != nil {
+				if err = os.Mkdir(out, 0777); err != nil {
```

Comment:
Optional: consider using MkdirAll that gives you Stat for free.

Reply:
Fixed in forthcoming push.